### PR TITLE
Support sub-techniques for newer API versions

### PR DESCRIPTION
**Pull request by Konverto-MartinGasser**
With commit [3f4ed3c](https://github.com/f-bader/SentinelARConverter/commit/3f4ed3ce6e5a1346042f03277e430b24cd6494b1) MITRE sub-techniques should be supported and do not require to be removed anymore.

### CODE REVIEW

**Comment by f-bader** [comment]
I guess this is only true if the api version is current enough. See #37 
My solution would be to validate that the selected api version is above this threshold and only then allow subtechniques

**Comment by f-bader** [comment]
Double checked the documentation and it's not as easy as to just keep them. There needs to be the container with, and a container without the subrechniques.

Minimum API version is indeed [`2023-12-01-preview`](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2023-12-01-preview/AlertRules.json)

```json
"tactics": [
    "Reconnaissance"
],
"techniques": [
    "T1589",
    "T1592",
    "T1598"
],
"subTechniques": [
    "T1589.001",
    "T1592.001"
]
```

@Konverto-MartinGasser do you feel up for the task adding this including tests or rather wait until I have the time to implement it?

**Comment by Konverto-MartinGasser** [comment]
I was able to implement the function and do some local testing/deploying, which was working as expected. Tomorrow I will try to create matching tests, but honestly, I don't think that I succeed. I'll let you know how it went.

**Comment by f-bader** [comment]
Very cool. If you need help with the tests let me know. Happy to help

**Comment by Konverto-MartinGasser** [comment]
Hey @f-bader, could you kindly take a look at my changes and especially on those tests I've written? I've never worked with test-files so far so I'm not sure I did the right thing ( at least they are not failing ;) )
I'm also not quite happy how I merge the properties relevantTechniques, subTechniques and techniques when converting from ARM to YAML. Maybe you have a suggestion on how to improve it?

Thanks

**Comment by f-bader** [comment]
Will do but most likely next week

**Review by f-bader — CHANGES_REQUESTED** [review]

File: src/public/Convert-SentinelARArmToYaml.ps1 [hunk 1]
```diff
@@ -343,7 +343,26 @@ function Convert-SentinelARArmToYaml {
...
                     }
-                    if ( -not $AnalyticsRuleCleaned.Contains($KeyName) ) {
+                    # Special case for subTechniques since we cannot add duplicate keys to $ValueNameMappingArm2Yaml
+                    # We must merge all techniques since (relevant)techniques could contain values not preset in subTechniques
+                    if ($PropertyName -like "*techniques") {
+                        foreach ($value in $AnalyticsRule.$PropertyName) {
+                            $KeyName = "techniques"
+                            $technique = $value -replace "(T\d{4})\.\d{3}", '$1'
+                            # Create an empty key
+                            if (-not($AnalyticsRuleCleaned[$KeyName])) {
```
Comment (f-bader) [inline src/public/Convert-SentinelARArmToYaml.ps1:353]:
```suggestion
                            if ( -not $AnalyticsRuleCleaned.Contains($KeyName) ) {
```

**Review by f-bader — APPROVED** [review]

**Comment by f-bader** [comment]
@Konverto-MartinGasser Amazing work and sorry it took me so long to validate and merge this.

**Pull request merged**